fix service issues on RHEL 7

larsks opened the pull request:
RHEL 7 uses systemd for managing services, so setting service_provider to
"redhat" by default results in an error.  This change (a) sets service_provider
to undef in all cases, and (b) exposes a service_provider parameter in init.pp
to allow this to be set if necessary by a calling manifest.

Closes #14.

soylent commented on manifests/params.pp:
```diff
@@ -37,14 +37,7 @@
-    $service_provider = 'redhat'
-  } else {
-    $service_provider = undef
-  }
```
I think we need to keep this hack for some time because the addressed issue has been resolved only in Puppet >= 4.5.0.

soylent commented on manifests/init.pp:
```diff
@@ -21,6 +21,7 @@
   $service_ensure = $::fluentd::params::service_ensure,
   $service_enable = $::fluentd::params::service_enable,
   $service_manage = $::fluentd::params::service_manage,
+  $service_provider = $::fluentd::params::service_provider,
```
Could you please add this param to the README

larsks commented on spec/classes/service_spec.rb:
```diff
@@ -2,7 +2,7 @@
 RSpec.describe 'fluentd::service' do
   context 'with redhat', :redhat do
-    it { is_expected.to contain_service('td-agent').with(provider: 'redhat') }
+    it { is_expected.to contain_service('td-agent').without(:provider) }
```
I guess if we are keeping the workaround in params.pp, this change should be revered.

larsks replied:
Or "reverted", even.

larsks commented:
I think this addresses your comments.  In addition to adding `service_provider` to the README, I also added `plugin_provider`, since it wasn't there either.

soylent commented:
Thank you!
